zerver/tornado/views.py

```diff
@@ -128,4 +128,12 @@ def get_events_backend(request: HttpRequest, user_profile: UserProfile,
         return response
     if result["type"] == "error":
         raise result["exception"]
+
+    events: List[Dict[str, Any]] = []
+    for event in result['response']['events']:
+        if event['type'] == 'message' and 'internal_data' in event.keys():
+            del event['internal_data']
+        events.append(event)
+    result['response']['events'] = events
```

Comment:
I think if you did just:
```
    for event in result['response']['events']:
￼        if event['type'] == 'message' and 'internal_data' in event.keys():
￼            del event['internal_data']
```

The result would be the same.

That said, I think that this process is incorrect in a nasty/subtle way -- we're mutating the original event data structure maintained in `event_queue.py`, and thus effectively deleting `internal_data` for everyone, not just this response.  I think the fix is to do the code as written with an added `event.copy()`.  It might be worth moving this logic into a function and writing a couple tests for it, to make sure we don't accidentally refactor such a bug back into existence.

Reply:
Totally forgot the `copy` part. Sorry for that. Also I think `deepcopy` would be better.